zkevm-circuits/src/evm_circuit/util/constraint_builder.rs

```diff
@@ -1024,19 +1042,29 @@ impl<'a, F: FieldExt> ConstraintBuilder<'a, F> {
     }
 
     pub(crate) fn add_constraints(&mut self, constraints: Vec<(&'static str, Expression<F>)>) {
-        self.cb.add_constraints(constraints);
+        for (name, constraint) in constraints {
+            self.add_constraint(name, constraint);
+        }
     }
 
     pub(crate) fn add_constraint(&mut self, name: &'static str, constraint: Expression<F>) {
-        self.cb.add_constraint(name, constraint);
+        let constraint = match &self.condition {
+            Some(condition) => condition.clone() * constraint,
+            None => constraint,
+        };
+
+        let constraint = self.split_expression(name, constraint, MAX_DEGREE - 2);
+
+        self.validate_degree(constraint.degree(), name);
+        self.constraints.push((name, constraint));
     }
 
     pub(crate) fn add_constraint_first_step(
         &mut self,
         name: &'static str,
         constraint: Expression<F>,
     ) {
```

Comment:
It seems we can also remove the `+ 1`  due to `Add 1 more degree due to the selector` below, I think it was a mistake by me.